src/types/bigint.rs

```diff
@@ -0,0 +1,107 @@
+use super::{private::ValueInternal, Value};
+use crate::context::internal::Env;
+use crate::context::Context;
+use crate::handle::{internal::TransparentNoCopyWrapper, Handle, Managed};
+use neon_runtime;
+use neon_runtime::raw;
+use std::fmt;
+use std::fmt::Debug;
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+#[cfg_attr(docsrs, doc(cfg(feature = "napi-6")))]
+pub struct JsBigInt(raw::Local);
+
+impl Value for JsBigInt {}
+
+unsafe impl TransparentNoCopyWrapper for JsBigInt {
+    type Inner = raw::Local;
+
+    fn into_inner(self) -> Self::Inner {
+        self.0
+    }
+}
+
+impl Managed for JsBigInt {
+    fn to_raw(&self) -> raw::Local {
+        self.0
+    }
+
+    fn from_raw(_: Env, h: raw::Local) -> Self {
+        JsBigInt(h)
+    }
+}
+
+#[derive(Debug)]
+pub struct GetBigIntLossyValueResult<T> {
```

Comment:
It's fairly common in Rust to put the data in the `Err` variant when it is still usable after a partial failure. For example, see [`LockResult`](https://doc.rust-lang.org/std/sync/type.LockResult.html) on `Mutex`.

I think it would be more idiomatic to create an `Error` wrapper (mostly for clear printing and `?` error propagation) that indicates data loss. If the user doesn't mind the loss, they can add `.unwrap_or_else(Error::into_inner)`.

```rust
struct Error<T>(T);

impl<T> Error<T> {
    fn into_inner(self) -> T {
        self.0
    }
}

impl<T> std::error::Error for Error {}
```